### PR TITLE
ThemeEditor: Redesign the UI to be comfier to use

### PR DESCRIPTION
**Pull request by AtkinsSJ**
Before:
ThemeEditor-old

After:
image

The old UI worked, and was basically effortless to add new theme properties to, but it made it hard to compare and modify different values together. So, this is an attempt to make that easier. :^)

This is a draft since it relies on a couple of fixes by Karol: #13945 and #13997. Otherwise the scrolling panel *really* doesn't work.

I planned on also adding tabs to the preview area so you could switch between multiple previews covering a wider variety of things, but this PR is already pretty big and that was turning out to be less than straightforward. So it'll come later. :^)

### CODE REVIEW

File: Userland/Applications/ThemeEditor/MainWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+
+    auto achromatopsia_accessibility_action = GUI::Action::create_checkable("Achromatopsia", { Mod_AltGr, Key_8 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_achromatopsia());
+    });
+
+    auto achromatomaly_accessibility_action = GUI::Action::create_checkable("Achromatomaly", { Mod_AltGr, Key_9 }, [&](auto&) {
+        m_preview_widget->set_color_filter(Gfx::ColorBlindnessFilter::create_achromatomaly());
+    });
+
+    auto preview_type_action_group = make<GUI::ActionGroup>();
```
Comment (krkk) [inline Userland/Applications/ThemeEditor/MainWidget.cpp:298]: This probably shouldn’t be a local object. The actions are not exclusive.

**Comment by filiphsandstrom** [comment]
I'd move the checkboxes to the right of the text so it matches the dropdown lists, other than that this looks great!

File: Userland/Applications/ThemeEditor/MainWidget.cpp (hunk 1 above)
Comment (AtkinsSJ) [reply Userland/Applications/ThemeEditor/MainWidget.cpp:298]: Thanks! I assumed the ActionGroup would be kept alive by something but it's not.

**Comment by AtkinsSJ** [comment]
> I'd move the checkboxes to the right of the text so it matches the dropdown lists, other than that this looks great!

Good idea! This is something I wanted on another app but LibGUI didn't support it... turned out to only take 15 minutes to add. :sweat_smile:

**Comment by AtkinsSJ** [comment]
:tada: Karol's PRs got merged, so let's undraft this! :^)

There is still one known bug - the property tabs that don't start selected, calculate their content width without including the scrollbar for some reason. It's still usable, and the margins mean you can even still see everything, you just get an unnecessary horizontal scrollbar.

**Comment by linusg** [comment]
This has a conflict

**Comment by AtkinsSJ** [comment]
> This has a conflict

Thanks! I've fixed that.

**Pull request merged**